python/pyspark/sql/dataframe.py

```diff
@@ -1223,10 +1226,8 @@ def withColumnRenamed(self, existing, new):
         >>> df.withColumnRenamed('age', 'age2').collect()
         [Row(age2=2, name=u'Alice'), Row(age2=5, name=u'Bob')]
         """
-        cols = [Column(_to_java_column(c)).alias(new)
-                if c == existing else c
-                for c in self.columns]
-        return self.select(*cols)
+        assert existing in self.columns, "%s is not an existing column" % existing
```

Comment:
We already did this for some of them, for example, AnalysisException and IllegalArgumentException, these could help.